compiler/ballerina-lang/src/main/java/io/ballerina/projects/PackageResolution.java

```diff
@@ -160,14 +161,18 @@ private boolean getSticky(PackageContext rootPackageContext) {
             this.autoUpdate = false;
             return true;
         }
+
         // set sticky if `build` file exists and `last_update_time` not passed 24 hours
         if (rootPackageContext.project().kind() == ProjectKind.BUILD_PROJECT) {
             Path buildFilePath = this.rootPackageContext.project().sourceRoot().resolve(TARGET_DIR_NAME)
                     .resolve(BUILD_FILE);
             if (Files.exists(buildFilePath) && buildFilePath.toFile().length() > 0) {
                 try {
                     BuildJson buildJson = readBuildJson(buildFilePath);
-                    if (buildJson != null && !buildJson.isExpiredLastUpdateTime()) {
+                    // if distribution is not same, we anyway return sticky as false
+                    if (buildJson != null &&
+                            buildJson.distributionVersion().equals(RepoUtils.getBallerinaShortVersion()) &&
```

Comment:
dist version can be null if `target/build` is generated with `beta3`. Is this handled?

Reply:
Added null check

Reply:
Shall we add a test case for this?